CRM Solution Manager/CRM Solution/CrmSolutionHelper.cs

```diff
@@ -319,6 +327,135 @@ private void ExportListOfSolutionsToBeMerged(OrganizationServiceProxy servicePro
+                    clientCredentials.UserName.UserName = instance.Attributes["syed_name"].ToString();
+                    clientCredentials.UserName.Password = instance.Attributes["syed_password"].ToString();
+                    OrganizationServiceProxy client = new OrganizationServiceProxy(new Uri(instance.Attributes["syed_instanceurl"].ToString()), null, clientCredentials, null);
+                    ImportSolution(client, solutionFile.SolutionFilePath ?? solutionFile.SolutionFilePathManaged);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Method import master solution to deployment instance
+        /// </summary>
+        /// <param name="serviceProxy">organization service proxy</param>
+        /// <param name="solutionImportPath">solution import path</param>
+        public static void ImportSolution(OrganizationServiceProxy serviceProxy, string solutionImportPath)
+        {
+            byte[] fileBytes = File.ReadAllBytes(solutionImportPath);
+
+            ImportSolutionRequest impSolReq = new ImportSolutionRequest()
+            {
+                CustomizationFile = fileBytes,
+                ImportJobId = Guid.NewGuid(),
+                OverwriteUnmanagedCustomizations = true,
+                SkipProductUpdateDependencies = true,
+                PublishWorkflows = false,
+            };
+            ExecuteAsyncRequest importRequest = new ExecuteAsyncRequest()
+            {
+                Request = impSolReq
+            };
+            ExecuteAsyncResponse importRequestResponse = (ExecuteAsyncResponse)serviceProxy.Execute(importRequest);
+
+            Console.WriteLine("Started importing solution");
+
+            // put in sleep for every 30 seconds to check the status of import using asyncoperation entity
+            string solutionImportResult = null;
+            while (solutionImportResult == null)
+            {
+                System.Threading.Thread.Sleep(30000);
+                Console.WriteLine("still importing......");
+                Guid asyncJobId = importRequestResponse.AsyncJobId;
+                Entity job = (Entity)serviceProxy.Retrieve("asyncoperation", asyncJobId, new ColumnSet(new System.String[] { "asyncoperationid", "statuscode", "message" }));
+                int jobStatusCode = ((OptionSetValue)job["statuscode"]).Value;
+                switch (jobStatusCode)
+                {
+                    // Success
+                    case 30:
+                        solutionImportResult = "success";
+                        Console.WriteLine("Solution imported successfully");
+                        break;
+                    //Pausing  
+                    case 21:
+                        break;
+                    //Canceling
+                    case 22:
+                        break;
+                    //Failed
+                    case 31:
+                        break;
+                    //Canceled
+                    case 32:
```

Comment:
for case 21,22,31 log in console log, abort the entire execution if it fails here.